backend/models/user.js

```diff
@@ -74,12 +91,57 @@ const findOne = async function (query, projection) {
 	return await DB.findOne("admin", COLL_NAME, query, projection);
 };
 
+const handleAuthenticateFail = async function (username) {
+	const currentTime = Date.now();
+
+	try {
+		const user = await User.findByUserName(username);
```

Comment:
I don't think we need to do any of that here. All we need to do is increment the failedLoginCount and update the last time we tried it? I don't think you need to do a read at all prior to the update.

You can use `$inc` to increment something https://docs.mongodb.com/manual/reference/operator/update/inc/

And do a `findAndUpdate` to update the document and get the counter back out all in one query